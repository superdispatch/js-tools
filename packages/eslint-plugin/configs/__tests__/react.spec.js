'use strict';

const { testInheritance } = require('../__testutils__/test-eslint-config');

testInheritance('react', 'base');
