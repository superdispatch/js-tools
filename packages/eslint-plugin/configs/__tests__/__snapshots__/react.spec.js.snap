// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inheritance config 1`] = `
Snapshot Diff:
- First value
+ Second value

  Object {
-   "env": Object {},
+   "env": Object {
+     "browser": true,
+   },
    "globals": Object {},
    "parser": null,
    "parserOptions": Object {
+     "ecmaFeatures": Object {
+       "jsx": true,
+     },
      "ecmaVersion": 2018,
    },
    "plugins": Array [
      "@superdispatch",
      "prettier",
      "import",
+     "react-hooks",
+     "react",
    ],
    "settings": Object {},
  }
`;

exports[`Inheritance dev rules 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`Inheritance rules 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ --- --- @@
  Object {
+   "@superdispatch/jsx-no-spread-object-expression": Array [
+     "error",
+   ],
    "array-bracket-newline": Array [
@@ --- --- @@
    "quotes": Array [
+     0,
+   ],
+   "react-hooks/exhaustive-deps": Array [
+     "error",
+   ],
+   "react-hooks/rules-of-hooks": Array [
+     "error",
+   ],
+   "react/display-name": Array [
+     2,
+   ],
+   "react/jsx-boolean-value": Array [
+     "error",
+     "always",
+   ],
+   "react/jsx-child-element-spacing": Array [
+     "off",
+   ],
+   "react/jsx-closing-bracket-location": Array [
+     "off",
+   ],
+   "react/jsx-closing-tag-location": Array [
+     "off",
+   ],
+   "react/jsx-curly-newline": Array [
+     "off",
+   ],
+   "react/jsx-curly-spacing": Array [
+     "off",
+   ],
+   "react/jsx-equals-spacing": Array [
+     "off",
+   ],
+   "react/jsx-first-prop-new-line": Array [
+     "off",
+   ],
+   "react/jsx-indent": Array [
+     "off",
+   ],
+   "react/jsx-indent-props": Array [
+     "off",
+   ],
+   "react/jsx-key": Array [
+     2,
+   ],
+   "react/jsx-max-props-per-line": Array [
+     "off",
+   ],
+   "react/jsx-no-comment-textnodes": Array [
+     2,
+   ],
+   "react/jsx-no-duplicate-props": Array [
+     2,
+   ],
+   "react/jsx-no-target-blank": Array [
+     2,
+   ],
+   "react/jsx-no-undef": Array [
+     2,
+   ],
+   "react/jsx-one-expression-per-line": Array [
+     "off",
+   ],
+   "react/jsx-props-no-multi-spaces": Array [
+     "off",
+   ],
+   "react/jsx-space-before-closing": Array [
+     "off",
+   ],
+   "react/jsx-tag-spacing": Array [
+     "off",
+   ],
+   "react/jsx-uses-react": Array [
+     2,
+   ],
+   "react/jsx-uses-vars": Array [
+     2,
+   ],
+   "react/jsx-wrap-multilines": Array [
+     "off",
+   ],
+   "react/no-children-prop": Array [
+     2,
+   ],
+   "react/no-danger-with-children": Array [
+     "error",
+   ],
+   "react/no-deprecated": Array [
+     2,
+   ],
+   "react/no-direct-mutation-state": Array [
+     2,
+   ],
+   "react/no-find-dom-node": Array [
+     2,
+   ],
+   "react/no-is-mounted": Array [
+     2,
+   ],
+   "react/no-render-return-value": Array [
+     2,
+   ],
+   "react/no-string-refs": Array [
+     2,
+   ],
+   "react/no-unescaped-entities": Array [
+     2,
+   ],
+   "react/no-unknown-property": Array [
+     2,
+   ],
+   "react/no-unsafe": Array [
      0,
+   ],
+   "react/prop-types": Array [
+     "off",
+   ],
+   "react/react-in-jsx-scope": Array [
+     2,
+   ],
+   "react/require-render-return": Array [
+     2,
    ],
`;
